src/Uuid/Uuid.php

```diff
@@ -0,0 +1,352 @@
+
+            return null;
+        }
+
+        if (null === $parsed = self::uuid_parse_as_array($uuid)) {
+            return false;
+        }
+
+        $uuid = str_replace('-', '', $uuid);
+
+        return pack('H*', $uuid);
+    }
+
+    public static function uuid_unparse($uuidAsBinary)
+    {
+        if (!\is_string($uuidAsBinary)) {
+            trigger_error(sprintf("uuid_unparse() expects parameter 1 to be string, %s given", gettype($uuid)), E_USER_WARNING);
+
+            return null;
+        }
+
+        $dataAsArray = unpack('H*', $uuidAsBinary);
+        $data = $dataAsArray[1];
+
+        if (32 != \strlen($data)) {
+            return false;
+        }
+
+        $uuid = sprintf('%s-%s-%s-%s-%s',
+            substr($data, 0, 8),
+            substr($data, 8, 4),
+            substr($data, 12, 4),
+            substr($data, 16, 4),
+            substr($data, 20, 12)
+        );
+
+        if (null === self::uuid_parse_as_array($uuid)) {
+            return false;
+        }
+
+        return $uuid;
+    }
+
+    private static function uuid_generate_random()
+    {
+        return sprintf('%08x-%04x-%04x-%04x-%06x%06x',
+            // 32 bits for "time_low"
+            random_int(0, 0xffffffff),
+
+            // 16 bits for "time_mid"
+            random_int(0, 0xffff),
+
+            // 16 bits for "time_hi_and_version",
+            // four most significant bits holds version number 4
+            random_int(0, 0x0fff) | 0x4000,
+
+            // 16 bits:
+            // * 8 bits for "clk_seq_hi_res",
+            // * 8 bits for "clk_seq_low",
+            // two most significant bits holds zero and one for variant DCE1.1
+            random_int(0, 0x3fff) | 0x8000,
+
+            // 48 bits for "node"
+            random_int(0, 0xffffff),
+            random_int(0, 0xffffff)
+        );
+    }
+
+    /**
+     * @see http://tools.ietf.org/html/rfc4122#section-4.2.2
+     */
+    private static function uuid_generate_time()
+    {
+        // https://tools.ietf.org/html/rfc4122#section-4.1.4
+        // 0x01b21dd213814000 is the number of 100-ns intervals between the
+        // UUID epoch 1582-10-15 00:00:00 and the Unix epoch 1970-01-01 00:00:00.
+        $offset = 0x01b21dd213814000;
+        $timeOfDay = gettimeofday();
+        $time = ($timeOfDay['sec'] * 10000000) + ($timeOfDay['usec'] * 10) + $offset;
+
+        // https://tools.ietf.org/html/rfc4122#section-4.1.5
+        // We are using a random data for the sake of simplicity: since we are
+        // not able to get a super precise timeOfDay as a unique sequence
+        $clockSeq = random_int(0, 0x3fff);
+
+        static $node;
+        if (null === $node) {
+            if (\function_exists('apcu_fetch')) {
+                $node = apcu_fetch('__symfony_uuid_node');
+                if (false === $node) {
+                    $node = sprintf('%06x%06x',
+                        random_int(0, 0xffffff),
```

Comment:
First of all, thanks for the nice polyfill. Having UUID generation directly intergrated is great :+1:

Just a small note for the node value: According to [RFC4122](https://tools.ietf.org/html/rfc4122#section-4.1.6), the MAC address [multicast bit](https://www.geeksforgeeks.org/introduction-of-mac-address-in-computer-network/) must be set if randomly or pseudo-randomly generated node values are used.

Reply:
Hello @xelan

Thanks :) 

 What does that mean exactly? What is the mylticasst bit?

Reply:
Hi @lyrixx! Thanks for your prompt response :) According to the RFC, the UUIDs generated with random node values must be distinguishable from regular MACs via setting that bit  to 1 (LSB of first MAC octet according to the linked source). Probably it would also make sense to set the locally administered bit to 1, as the random node values are not supposed to be globally unique.

Reply:
[Here ](https://tools.ietf.org/html/rfc4122#section-4.5)the reasoning behind that requirement is described. Based on that, I'd guess that setting both the multicast bit and the local bit would be the best solution.

Reply:
done, Thanks